src/Interpreters/ExpressionJIT.cpp

```diff
@@ -316,28 +316,6 @@ static bool isCompilableConstant(const ActionsDAG::Node & node)
     return node.column && isColumnConst(*node.column) && canBeNativeType(*node.result_type) && node.allow_constant_folding;
 }
 
-static bool checkIfFunctionIsComparisonEdgeCase(const ActionsDAG::Node & node, const IFunctionBase & impl)
```

Comment:
This should be done by the expression analyzer at about the same stage where it evaluates constants.